packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1283,6 +1293,40 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     };
   }
 
+  private configureSnapStart(props: FunctionProps): CfnFunction.SnapStartProperty | undefined {
+    // return/exit if no snapStart included
+    if (!props.snapStart) {
+      return undefined;
+    }
+
+    // SnapStart does not support arm64 architecture, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB.
+    // SnapStart doesn't support provisioned concurrency either, but that's configured at the version level,
+    // so it can't be checked at function set up time
+    // SnapStart supports the Java 11 and Java 17 (java11 and java17) managed runtimes.
+    // See https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html
+    if (props.snapStart != SnapStartConfig.ON_PUBLISHED_VERSIONS) {
+      throw new Error('SnapStart currently only support published versions');
+    }
+
+    if (props.runtime != Runtime.JAVA_11 && props.runtime != Runtime.JAVA_17 ) {
+      throw new Error('SnapStart currently only support Java 11/Java 17 runtime');
+    }
+
+    if (props.architecture == Architecture.ARM_64) {
+      throw new Error('SnapStart is currently not supported on Arm_64');
+    }
+
+    if (props.filesystem) {
+      throw new Error('SnapStart is currently not supported using EFS');
+    }
+
+    if (props.ephemeralStorageSize && props.ephemeralStorageSize?.toMebibytes() > 512) {
+      throw new Error('SnapStart is currently not supported using more than 512 MiB Ephemeral Storage');
+    }
+
+    return props.snapStart.snapStartConfig;
```

Comment:
Given my below comment we could the have this be - `return props.snapStart._render()`